server/controllers/stock/index.js

```diff
@@ -284,14 +297,39 @@ async function createInventoryAdjustment(req, res, next) {
     const period = await Fiscal.lookupFiscalYearByDate(new Date(movement.date));
     const periodId = period.id;
 
-    // pass reverse operations
     const trx = db.transaction();
-
     const uniqueAdjustmentUuid = uuid();
 
     let countNeedIncrease = 0;
     let countNeedDecrease = 0;
 
+    lots.forEach(lot => {
+      lot.quantityAvailable = 0;
+
+      if (lot.oldQuantity > lot.quantity) {
+        lot.isExit = 1;
+        lot.outputQuantity = lot.oldQuantity - lot.quantity;
+
+        if (stockAvailable) {
+          stockAvailable.forEach(stock => {
+            if (stock.uuid === lot.uuid) {
+              lot.quantityAvailable = stock.quantity;
+            }
+          });
+        }
+      }
+    });
+
+    filteredInvalidData = await lots.filter(l => l.outputQuantity > l.quantityAvailable);
+
+    if (filteredInvalidData.length) {
+      throw new BadRequest(
+        `There are stock movements,
```

Comment:
Are these duplicate error messages?

Reply:
Yes, and it is so in several uses of  ````throw new BadRequest````
![image](https://user-images.githubusercontent.com/8035588/112832911-e1527c80-908d-11eb-9df9-0596aaac4b83.png)